opennlp-tools/src/main/java/opennlp/tools/formats/BioNLP2004NameSampleStreamFactory.java

```diff
@@ -30,55 +29,56 @@
 /**
  * @see BioNLP2004NameSampleStream
  */
-public class BioNLP2004NameSampleStreamFactory<P> extends AbstractSampleStreamFactory<NameSample, P> {
+public class BioNLP2004NameSampleStreamFactory extends
+        AbstractSampleStreamFactory<NameSample, BioNLP2004NameSampleStreamFactory.Parameters> {
 
-  interface Parameters extends BasicFormatParams {
+  public interface Parameters extends BasicFormatParams {
     @ParameterDescription(valueName = "DNA,protein,cell_type,cell_line,RNA")
     String getTypes();
   }
 
   public static void registerFactory() {
     StreamFactoryRegistry.registerFactory(NameSample.class,
-        "bionlp2004", new BioNLP2004NameSampleStreamFactory<>(Parameters.class));
+        "bionlp2004", new BioNLP2004NameSampleStreamFactory(Parameters.class));
   }
 
-  protected BioNLP2004NameSampleStreamFactory(Class<P> params) {
+  protected BioNLP2004NameSampleStreamFactory(Class<Parameters> params) {
     super(params);
   }
 
   @Override
   public ObjectStream<NameSample> create(String[] args) {
-
-    Parameters params = ArgumentParser.parse(args, Parameters.class);
+    Parameters params = validateBasicFormatParameters(args, Parameters.class);
 
     int typesToGenerate = 0;
-
-    if (params.getTypes().contains("DNA")) {
+    String types = params.getTypes();
```

Comment:
Is there a reason for removing the `else` cascade here? Previous code seemed to check only once, if a match occured. New code will trigger subsequent `contains` ?

Reply:
Previous code was buggy. `BioNLP2004NameSampleStreamFactoryTest` shows why: multiple types in the parameter list would not be processed correctly. Now they are "all" included.

Reply:
Too much protein ^ :P

Reply:
Side note: 
Each `AbstractSampleStreamFactoryTest`'s primary idea is to check the **creation** of the corresponding XYSampleStream. For more detailed checks on the outcomes, it's better to adapt / enhance the test of the actual processing code, which resides outside the ...Factory classes.